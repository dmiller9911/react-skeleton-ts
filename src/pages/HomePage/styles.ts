import { StyleSheet } from 'aphrodite';

export default StyleSheet.create({
  contentHeader: {
    textAlign: 'center'
  },
  heroHeader: {
    color: '#fff',
    fontSize: '2.75rem',
    textAlign: 'center'
  }
});
