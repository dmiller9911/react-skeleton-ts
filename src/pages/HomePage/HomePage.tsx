import { Container, Heading, headingLevel } from '../../atoms';
import { ITechProps } from '../../molecules';
import { Hero, TechList} from '../../organisms';
import { HeroPage } from '../../templates';
import styles from './styles';
import * as React from 'react';

const techItems: ITechProps[] = [
  {
    description: 'TypeScript is a typed superset of JavaScript that compiles to plain JavaScript.',
    githubLink: 'https://github.com/Microsoft/TypeScript',
    iconUrl: require<string>('../../assets/techLogos/ts.png'),
    name: 'TypeScript'
  },
  {
    description: 'A JavaScript library for building user interfaces',
    githubLink: 'https://github.com/facebook/react',
    iconUrl: require<string>('../../assets/techLogos/react.png'),
    name: 'React',
  },
  {
    description: 'Declarative routing for React',
    githubLink: 'https://github.com/ReactTraining/react-router',
    iconUrl: require<string>('../../assets/techLogos/react-router.png'),
    name: 'React Router (V3)'
  },
  {
    description: `It's inline styles, but they work!`,
    githubLink: 'https://github.com/Khan/aphrodite',
    iconUrl: require<string>('../../assets/techLogos/aphrodite.png'),
    name: 'Aphrodite'
  },
  {
    description: 'A bundler for javascript and friends.',
    githubLink: 'https://github.com/webpack/webpack',
    iconUrl: require<string>('../../assets/techLogos/webpack.png'),
    name: 'Webpack'
  },
  {
    description: 'Painless JavaScript Testing.',
    githubLink: 'https://github.com/facebook/jest',
    iconUrl: require<string>('../../assets/techLogos/jest.png'),
    name: 'Jest'
  },
];

export default function HomePage(props: {}) {
  const HeroComponent = (
    <Hero>
      <Container>
        <Heading level={ headingLevel.h1} classNames={ [styles.heroHeader] }>
          React + TypeScript
        </Heading>
      </Container>
    </Hero>
  );

  return (
    <HeroPage hero={ HeroComponent }>
      <section>
        <Heading level={ headingLevel.h2 } classNames={ [styles.contentHeader] }>
          Basic Stack
        </Heading>
        <div>
          <TechList techItems={ techItems } />
        </div>
      </section>
    </HeroPage>
  );
}
