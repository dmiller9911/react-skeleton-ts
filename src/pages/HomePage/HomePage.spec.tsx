jest.unmock('./HomePage');

import HomePage from './HomePage';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';

describe('HomePage', () => {
  it('matches snapshot', () => {
    expect(shallowToJson(shallow(<HomePage />))).toMatchSnapshot();
  });
});
