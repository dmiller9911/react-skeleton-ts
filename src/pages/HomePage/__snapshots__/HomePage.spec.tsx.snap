exports[`HomePage matches snapshot 1`] = `
<HeroPage
  hero={
    <Hero>
      <Container>
        <Heading
          classNames={
            Array [
              Object {
                "_definition": Object {
                  "color": "#fff",
                  "fontSize": "2.75rem",
                  "textAlign": "center",
                },
                "_name": "heroHeader_1wu7yln",
              },
            ]
          }
          level={1}>
          React + TypeScript
        </Heading>
      </Container>
    </Hero>
  }>
  <section>
    <Heading
      classNames={
        Array [
          Object {
            "_definition": Object {
              "textAlign": "center",
            },
            "_name": "contentHeader_1ntyrs8",
          },
        ]
      }
      level={2}>
      Basic Stack
    </Heading>
    <div>
      <TechList
        techItems={
          Array [
            Object {
              "description": "TypeScript is a typed superset of JavaScript that compiles to plain JavaScript.",
              "githubLink": "https://github.com/Microsoft/TypeScript",
              "iconUrl": "/home/douglasmi/src/react-skeleton-ts/src/assets/techLogos/ts.png",
              "name": "TypeScript",
            },
            Object {
              "description": "A JavaScript library for building user interfaces",
              "githubLink": "https://github.com/facebook/react",
              "iconUrl": "/home/douglasmi/src/react-skeleton-ts/src/assets/techLogos/react.png",
              "name": "React",
            },
            Object {
              "description": "Declarative routing for React",
              "githubLink": "https://github.com/ReactTraining/react-router",
              "iconUrl": "/home/douglasmi/src/react-skeleton-ts/src/assets/techLogos/react-router.png",
              "name": "React Router (V3)",
            },
            Object {
              "description": "It\'s inline styles, but they work!",
              "githubLink": "https://github.com/Khan/aphrodite",
              "iconUrl": "/home/douglasmi/src/react-skeleton-ts/src/assets/techLogos/aphrodite.png",
              "name": "Aphrodite",
            },
            Object {
              "description": "A bundler for javascript and friends.",
              "githubLink": "https://github.com/webpack/webpack",
              "iconUrl": "/home/douglasmi/src/react-skeleton-ts/src/assets/techLogos/webpack.png",
              "name": "Webpack",
            },
            Object {
              "description": "Painless JavaScript Testing.",
              "githubLink": "https://github.com/facebook/jest",
              "iconUrl": "/home/douglasmi/src/react-skeleton-ts/src/assets/techLogos/jest.png",
              "name": "Jest",
            },
          ]
        } />
    </div>
  </section>
</HeroPage>
`;
